### DISCUSSION TITLE
I'd love to be able to use the `?` (question mark) operator with JSON

### DISCUSSION THREAD

**Discussion by samuelcolvin** [duckdb/duckdb#11559] [Ideas]
(Appologies if this has been asked for before, I searched issues and discussion and couldn't find, it but I'm aware that might be before "?" is hard to search for)

I'd love to be able to use `json_haystack ? 'key_needle'` like [in postgres](https://www.postgresql.org/docs/16/functions-json.html#FUNCTIONS-JSONB-OP-TABLE).

Currently instead I seem to need to use

```sql
list_contains(json_keys(json_haystack), 'key_needle')
```

This is the main problem I've found with JSON querying in duckdb, but I guess the other advanced JSON operators like `?|` and `?&` [etc.](https://www.postgresql.org/docs/16/functions-json.html#FUNCTIONS-JSONB-OP-TABLE) might be good too.

**Comment by samuelcolvin** [comment]
Please excuse the ping, @Mytherin what do you think? Is there any chance of this being added?

**Reply by Mytherin** [reply]
I've had a look at this and I think the chance is small - at least until we get a new parser. DuckDB supports using the question mark (`?`) as a parameter natively which means it cannot also be used as an operator very easily at the parser layer.
